relay-wstring/src/lib.rs

```diff
@@ -42,15 +45,17 @@ impl fmt::Display for Utf16Error {
 impl Error for Utf16Error {}
 
 impl Utf16Error {
-    /// Create a new [Utf16Error].
-    pub fn new() -> Self {
-        Self {}
+    /// Returns the index in given bytes up to which valid UTF-16 was verified.
+    pub fn valid_up_to(&self) -> usize {
+        self.valid_up_to
     }
-}
 
-impl Default for Utf16Error {
-    fn default() -> Self {
-        Self::new()
+    /// Return the length of the error if it might be recoverable.
+    ///
+    /// If `Some` you should be able to attempt to parse again from the offset given by
+    /// adding the error legnth to [Self::valid_up_to].
+    pub fn error_len(&self) -> Option<usize> {
```

Comment:
Can you copy over semantics and description from [`Utf8Error::error_len`](https://doc.rust-lang.org/stable/std/str/struct.Utf8Error.html#method.error_len)? When reading this description, I did not understand why there could be `None` and why `Some`.